app/assets/javascripts/discourse/app/instance-initializers/safari-viewport-setup.js

```diff
@@ -0,0 +1,20 @@
+/*
+  In iOS Safari, setting user-scalable=no doesn't actually prevent the user from zooming in.
+  But, it does prevent the annoying 'auto zoom' when focussing input fields with small font-sizes.
+*/
+export default {
+  initialize(container) {
+    if (!container.lookup("service:capabilities").isIOS) {
+      return;
+    }
+
+    const viewport = document.querySelector("meta[name=viewport]");
+    if (viewport) {
+      const current = viewport.getAttribute("content");
+      viewport.setAttribute(
+        "content",
+        current.replace("user-scalable=yes", "user-scalable=no")
```

Comment:
I can live with this... might become an issue if one day Safari decides to reverse ignoring this viewport property. 

Safari also ignores `interactive-widget` from the viewport meta tag, and outputs a console error: 

![CleanShot 2025-01-20 at 14 08 16@2x](https://github.com/user-attachments/assets/3be1bbee-d0f4-40f2-b2c2-c958acb428be)

Now that we have this override for Safari iOS specifically, we can strip that too?

Reply:
Yeah, this PR is replacing one workaround with a different workaround. It's still not ideal, but at least with this one we don't have the visibly inconsistent sizing.

Unfortunately this initializer runs too late to avoid that `interactive-widget` warning. Technically we could move all this logic to the server, but I don't want to because it means we'd need to partition our anonymous cache based on iOS/non-iOS.

So I think the best strategy is to ignore the `interactive-widget` warning, and hope Safari adds support soon 🤞

Reply:
On second thoughts, we can move that `interactive-widget` value into the initializer as well: https://github.com/discourse/discourse/pull/30889